docs/book/v3/application-integration/usage-in-a-laminas-mvc-application.md

```diff
@@ -67,48 +67,43 @@ class AlbumForm extends Form implements InputFilterProviderInterface
 }
 ```
 
-## Using Form
+## Create Controller
 
-### Create Controller
-
-[Create a controller class](https://docs.laminas.dev/laminas-mvc/quick-start/#create-a-controller)
-and inject the form via the constructor, e.g.
-`module/Album/Controller/AlbumController.php`:
+[Create a controller class](https://docs.laminas.dev/laminas-mvc/quick-start/#create-a-controller) and inject the form element manager via the constructor, e.g. `module/Album/Controller/AlbumController.php`:
 
 ```php
 namespace Album\Controller;
 
 use Album\Form\AlbumForm;
-use Laminas\Form\FormInterface;
+use Laminas\Form\FormElementManager;
 use Laminas\Mvc\Controller\AbstractActionController;
 
 class AlbumController extends AbstractActionController
 {
-    /** @var FormInterface */
-    private $form;
-    
-    public function __construct(AlbumForm $form)
-    {
-        $this->form = $form;
-    }
+    public function __construct(
+        public readonly FormElementManager $formElementManager
+    ) {}
     
     public function addAction()
     {
+        /** @var AlbumForm $form */
+        $form = $this->formElementManager->get(AlbumForm::class);
```

Comment:
See https://github.com/doctrine/coding-standard/pull/47:
```suggestion
        $form = $this->formElementManager->get(AlbumForm::class);
        \assert($form instanceof AlbumForm);
```

Reply:
_We do not use the Doctrine coding standard._

That is correct, but the question is if it belongs in this tutorial? The one line raises more questions and therefore needs additional explanation. The problem is that we use it in the code of various components, but nowhere in the documentation.

Reply:
> _We do not use the Doctrine coding standard._

Neither are `readonly` attributes. My link was just about the *why* we should do it.

> The one line raises more questions and therefore needs additional explanation.

We have never explained `/** @var */` either.

> The problem is that we use it in the code of various components, but nowhere in the documentation.

`assert()` in the worst case scenario behaves exactly like `/** @var */`; in the best case scenario is a runtime type check.
Why not pushing new good practices over the worse old alternative?

Reply:
> We have never explained /** @var */ either.

This is nothing new in the world of PHP and also in the documentation.

> Why not pushing new good practices over the worse old alternative?

Please don't misunderstand me, I would like to take it over but we are still fighting and have to live with "worse old alternative" as well. Because we have to deal with the old, ugly and wrong description in the getting started tutorial. Therefore, a balance must be struck here.

_Give me some time to consider._

Reply:
@Slamdunk 
Thanks for review and input! 👍🏻

Reply:
@Slamdunk 
I knew I read it somewhere. 😃 

We already have this topic in the contributing guide:

> ### Performing assertions on the code
> We distinguish assertions in two categories:
>
> * Expectations of something that is always supposed to be true
> * Expectations of some input to follow some rule
>
> The former is a development-only concern, aiming to aid tools (e.g. IDEs, static analysis) to perform a more accurate type detection. In this scenario, you should use the [assert()](https://www.php.net/manual/en/function.assert.php) function, which executes the expression when [zend.assertions](https://www.php.net/manual/en/ini.core.php#ini.zend.assertions) is enabled. That will make sure that the condition is valid, while easing future refactoring.
>
> …

https://github.com/laminas/.github/blob/main/CONTRIBUTING.md#performing-assertions-on-the-code

Reply:
Nice, so we can use it :muscle: